packages/payments-client/src/core/private-route-wrapper.tsx

```diff
@@ -1,19 +1,49 @@
 import * as React from 'react'
+import { Route, Redirect, useLocation, Switch } from 'react-router'
 import { useReapitConnect } from '@reapit/connect-session'
 import { Loader, Section, FlexContainerResponsive, AppNavContainer, FlexContainerBasic } from '@reapit/elements'
 import Menu from '../components/ui/menu'
 import { reapitConnectBrowserSession } from './connect-session'
-import { Redirect, useLocation } from 'react-router'
 import Routes from '../constants/routes'
+import { catchChunkError } from './router'
 
 const { Suspense } = React
 
 export type PrivateRouteWrapperProps = {}
 
+const PaymentSessionPage = React.lazy(() => catchChunkError(() => import('../components/pages/payment-session')))
+
 export const PrivateRouteWrapper: React.FunctionComponent<PrivateRouteWrapperProps> = ({ children }) => {
-  const { connectSession, connectInternalRedirect } = useReapitConnect(reapitConnectBrowserSession)
   const location = useLocation()
-  const currentUri = `${location.pathname}${location.search}`
+  const { pathname, search } = location
+  const currentUri = `${pathname}${search}`
+  const queryParams = new URLSearchParams(search)
+  const session = queryParams.get('session')
```

Comment:
This logic can just be in the PaymentSession component > render the component if the route matches, if no session, just render a Helper component saying "you don't appear to have a valid session"